esdt-safe/src/lib.rs

```diff
@@ -280,14 +278,69 @@ pub trait EsdtSafe:
                 *burned += &actual_bridged_amount;
             });
         }
+
+        (
+            batch_id,
+            tx_nonce,
+            payment_token,
+            actual_bridged_amount,
+            required_fee,
+            to.as_managed_buffer().clone(),
+            caller.as_managed_buffer().clone(),
+        )
+    }
+
+    // endpoints
+
+    /// Create an MultiversX -> Ethereum transaction. Only fungible tokens are accepted.
+    ///
+    /// Every transfer will have a part of the tokens subtracted as fees.
+    /// The fee amount depends on the global eth_tx_gas_limit
+    /// and the current GWEI price, respective to the bridged token
+    ///
+    /// fee_amount = price_per_gas_unit * eth_tx_gas_limit
+    #[payable("*")]
+    #[endpoint(createTransaction)]
+    fn create_transaction(&self, to: EthAddress<Self::Api>) {
+        let (batch_id, tx_nonce, payment_token, actual_bridged_amount, required_fee, to, from) =
+            self.create_transaction_common(to);
         self.create_transaction_event(
             batch_id,
             tx_nonce,
             payment_token,
             actual_bridged_amount,
             required_fee,
-            tx.to,
-            tx.from,
+            to,
+            from,
+        );
+    }
+
+    #[payable("*")]
+    #[endpoint(createTransactionSCCall)]
+    fn create_transaction_sc_call(
+        &self,
+        to: EthAddress<Self::Api>,
+        data: ManagedBuffer<Self::Api>,
+    ) {
+        let (
+            batch_id,
```

Comment:
Same here. A type containing these items would be really useful for the readability.